### DISCUSSION TITLE
backup db with -stdout path results in zero size backup.sql file

### DISCUSSION THREAD

**Discussion by rabotond** [photoprism/photoprism#2054] [Q&A] [answered]
Hi,

I use Photoprism now quiet a long time and I now realized something went wrong as my usual db backups are creating zero sized backup files from the db even though the backup command says "-- Dump completed on 2022-02-20 16:24:34", and there is no error messages. It was working for quiet a long time but it now produces this issue. I use photoprism with docker-compose.

I used this command so far and it worked perfect:
`docker-compose exec photoprism photoprism backup -i - > backup_date.sql
`
I checked and if I dont specify the stdout path it works, the created db dump file in the container has size and indeed contains the db dump. However I want to save it outside of the container to my external hdd for backups so I specify always the -stdout path. 

Last time I remember what I changed and might cause the issue but not sure:
- updated photoprism
- updated my raspberry OS and the whole system(now it is on latest 64 bit raspberry os)

I managed to reproduce this issue also on my macbook and my raspberry as well. 

Your help is appreciated! 
Have a good day!

**Comment by lastzero** [comment, marked as answer]
Next time it's best to use GitHub Discussions for questions :)

The correct command is this one (note the "T" parameter):

```
docker-compose exec -T photoprism photoprism backup -i - > photoprism-db.sql
```

see https://docs.photoprism.app/getting-started/advanced/backups/
